chia/pools/pool_wallet.py

```diff
@@ -778,29 +798,54 @@ async def claim_pool_rewards(self, fee: uint64) -> TransactionRecord:
             self.log.info(
                 f"Farmer coin: {coin_record.coin} {coin_record.coin.name()} {coin_to_height_farmed[coin_record.coin]}"
             )
-        if len(all_spends) == 0:
+        if len(all_spends) == 0 or current_coin_record is None:
             raise ValueError("Nothing to claim, no unspent coinbase rewards")
 
-        # No signatures are required to absorb
-        spend_bundle: SpendBundle = SpendBundle(all_spends, G2Element())
+        claim_spend: SpendBundle = SpendBundle(all_spends, G2Element())
+
+        # If fee is 0, no signatures are required to absorb
+        full_spend: SpendBundle = claim_spend
+
+        if fee > 0:
+            get_max_send_amount = await self.standard_wallet.get_max_send_amount()
+            assert get_max_send_amount > fee
+            amount = uint64(0)
+            my_address = await self.standard_wallet.get_new_puzzlehash()
+            fee_tx = await self.standard_wallet.generate_signed_transaction(
+                amount,
+                my_address,
+                fee,
+                None,
+                None,
+                None,
+                False,
+                {Announcement(current_coin_record.coin.name(), b"$")},
+            )
+
+            full_spend = SpendBundle.aggregate([fee_tx.spend_bundle, claim_spend])
+            # The fee spend only, from the main wallet
+            await self.wallet_state_manager.add_pending_transaction(dataclasses.replace(fee_tx, spend_bundle=None))
+        assert full_spend.fees() == fee
 
+        current_time = uint64(int(time.time()))
+        # The claim spend, minus the fee amount from the main wallet
         absorb_transaction: TransactionRecord = TransactionRecord(
             confirmed_at_height=uint32(0),
-            created_at_time=uint64(int(time.time())),
+            created_at_time=current_time,
             to_puzzle_hash=current_state.current.target_puzzle_hash,
             amount=uint64(total_amount),
-            fee_amount=fee,
+            fee_amount=uint64(0),  # Notice that fee is zero here - is is accounted in self.standard_wallet
```

Comment:
You don't have to make this change.  The TX records are mostly just for reporting to the GUI what is true about the transaction.  It is going to be true that this transaction went through with that fee just as it is true that the fee transaction went through with that fee.  You don't need to worry about them adding up in any way.